runtime/taskcontext.go

```diff
@@ -195,6 +212,15 @@ func (c *TaskContext) NewLogReader() (io.ReadCloser, error) {
 	return c.logStream.NextReader()
 }
 
+func (c *TaskContext) ExtractLog() (ioext.ReadSeekCloser, error) {
+	file, err := os.Open(c.logLocation)
```

Comment:
It's necessary for retrying artifact uploading if I recall correctly.  It's been awhile since I worked on this.